src/components/ColorPicker/components/TextPicker/TextPicker.tsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+
+import {classNames} from '../../../../utilities/css';
+import {HSBColor} from '../../../../utilities/color-types';
+import {
+  normalizeValue,
+  expandHex,
+  hsbToHex,
+  rgbStringToHex,
+  nameToHex,
+  hexToHsb,
+} from '../../../../utilities/color-transformers';
+import {
+  isColorName,
+  isHexString,
+  isHashlessHex,
+  isRgbString,
+} from '../../../../utilities/color-validation';
+import {
+  withAppProvider,
+  WithAppProviderProps,
+} from '../../../../utilities/with-app-provider';
+import {TextField} from '../../../TextField';
+import {SwatchBackground} from '../SwatchBackground';
+import styles from '../../ColorPicker.scss';
+
+export interface TextPickerProps {
+  color: HSBColor;
+  allowAlpha?: boolean;
+  onChange(hex: string): void;
+}
+
+type CombinedProps = TextPickerProps & WithAppProviderProps;
+
+interface State {
+  valueHasBeenUpdatedOnce: boolean;
```

Comment:
We could probably have this as a class property rather than state unless we needed it to trigger a re-render?